command_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/keybase/go-logging"
+)
+
+func TestEmptyRunCommand(t *testing.T) {
+	log := logging.Logger{Module: "test"}
+	_, err := RunCommand("", nil, time.Second, log)
+	t.Logf("Error: %s", err)
+	if err == nil {
+		t.Fatal("Should have errored")
+	}
+}
+
+func TestInvalidRunCommand(t *testing.T) {
+	log := logging.Logger{Module: "test"}
+	_, err := RunCommand("invalidexecutable", nil, time.Second, log)
+	t.Logf("Error: %s", err)
+	if err == nil {
+		t.Fatal("Should have errored")
+	}
+}
+
+func TestRunCommandEcho(t *testing.T) {
+	log := logging.Logger{Module: "test"}
+	out, err := RunCommand("echo", []string{"arg1", "arg2"}, time.Second, log)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expected := "arg1 arg2\n"
+	if out != expected {
+		t.Errorf("Unexpected output: %q != %q", out, expected)
+	}
+}
+
+func TestRunCommandTimeout(t *testing.T) {
+	log := logging.Logger{Module: "test"}
+	_, err := RunCommand("sleep", []string{"10"}, time.Second, log)
+	if err == nil {
```

Comment:
TestTimeoutProcessKilled tries to verify that the process was killed...

Unfortunately go doesn't expose the state of a process unless you call Wait(), but Wait() error if it's not running, so I am calling kill and asserting an error occurs since you can't kill a non-existent process.